compactor_scheduler/src/local_scheduler.rs

```diff
@@ -156,11 +327,315 @@ mod tests {
+        let partition_ids = vec![partition1.partition.id, partition2.partition.id];
+
+        // two files on partition1, to be replaced by one compacted file
+        let file_builder = TestParquetFileBuilder::default().with_line_protocol("table1 load=1 11");
+        let file1_1 = partition1.create_parquet_file(file_builder.clone()).await;
+        let file1_2 = partition1.create_parquet_file(file_builder).await;
+
+        let config = LocalSchedulerConfig {
+            commit_wrapper: None,
+            partitions_source_config: PartitionsSourceConfig::Fixed(
+                partition_ids.into_iter().collect::<HashSet<PartitionId>>(),
+            ),
+            shard_config: None,
+        };
+
+        let scheduler = LocalScheduler::new(
+            config,
+            BackoffConfig::default(),
+            catalog.catalog(),
+            Arc::new(MockProvider::new(Time::MIN)),
+            Arc::new(metric::Registry::default()),
+            false,
+        );
+
+        (scheduler, file1_1, file1_2)
+    }
+
+    #[tokio::test]
+    #[should_panic]
+    async fn test_status_update_none_should_panic() {
+        test_helpers::maybe_start_logging();
+
+        let (scheduler, _, _) = create_scheduler_with_partitions().await;
+        let jobs = scheduler.get_jobs().await;
+
+        for job in jobs {
+            let commit_update = CommitUpdate {
+                partition_id: job.partition_id,
+                delete: vec![],
+                upgrade: vec![],
+                target_level: data_types::CompactionLevel::Final,
+                create: vec![],
+            };
+
+            let _ = scheduler
+                .job_status(CompactionJobStatus {
+                    job,
+                    status: CompactionJobStatusVariant::Update(commit_update),
+                })
+                .await;
+        }
+    }
+
+    #[tokio::test]
+    async fn test_status_update_replacement() {
+        test_helpers::maybe_start_logging();
+
+        let (scheduler, existing_1, existing_2) = create_scheduler_with_partitions().await;
+        let jobs = scheduler.get_jobs().await;
+        let job = jobs
+            .into_iter()
+            .find(|job| job.partition_id == existing_1.partition.partition.id)
+            .unwrap();
+
+        let created = ParquetFileBuilder::new(1002)
+            .with_partition(job.partition_id.get())
+            .build();
+
+        let commit_update = CommitUpdate {
+            partition_id: job.partition_id,
+            delete: vec![existing_1.into(), existing_2.into()],
+            upgrade: vec![],
+            target_level: data_types::CompactionLevel::Final,
+            create: vec![created.into()],
+        };
+
+        assert!(scheduler
+            .job_status(CompactionJobStatus {
+                job,
+                status: CompactionJobStatusVariant::Update(commit_update),
+            })
+            .await
+            .is_ok());
```

Comment:
I think it would be valuable to test here and in the other tests that send `CompactionJobStatusVariant::Update` that the `created` IDs come back in the response. [The `assert_matches!` macro](https://docs.rs/assert_matches/latest/assert_matches/macro.assert_matches.html) should make this nice! Something like:

```suggestion
        assert_matches!(scheduler
            .job_status(CompactionJobStatus {
                job,
                status: CompactionJobStatusVariant::Update(commit_update),
            })
            .await,
            Ok(CompactionJobStatusResult::UpdatedParquetFiles(result)) if result == &[created.id]
        );
```

Reply:
This is me being impatient. Should have slowed down and did this test suite properly. Thanks for pointing out -- fixing!